code/_helpers/lists.dm

```diff
@@ -772,7 +772,8 @@ var/global/list/json_cache = list()
 	try
 		if(isnull(global.json_cache[json_to_decode]))
 			global.json_cache[json_to_decode] = json_decode(json_to_decode)
-		. = global.json_cache[json_to_decode]
+		var/list/decoded = global.json_cache[json_to_decode]
+		return decoded?.Copy() || list() // To prevent cache mutation.
```

Comment:
Not an expert on the system, but aren't there valid nonlist outputs of `json_decode`, like a single string, number, or bool for example?